src/jmh/java/hu/akarnokd/comparison/scrabble/ShakespearePlaysScrabbleWithCyclopsReactOpt.java

```diff
@@ -49,145 +66,99 @@
     )
     @Fork(1)
     public List<Entry<Integer, List<String>>> measureThroughput() throws InterruptedException {
-
-        //  to compute the score of a given word
-        Function<Integer, Integer> scoreOfALetter = letter -> letterScores[letter - 'a'];
-
-        // score of the same letters in a word
-        Function<Entry<Integer, MutableLong>, Integer> letterScore =
-                entry ->
-                        letterScores[entry.getKey() - 'a'] *
-                        Integer.min(
-                                (int)entry.getValue().get(),
-                                scrabbleAvailableLetters[entry.getKey() - 'a']
-                            )
-                    ;
-
-
-        Function<String, ReactiveSeq<Integer>> toIntegerIx =
-                string -> chars(string);
-
-        // Histogram of the letters in a given word
-        Function<String, ReactiveSeq<HashMap<Integer, MutableLong>>> histoOfLetters =
-                word ->  toIntegerIx.apply(word)
-                            .scanLeft(new HashMap<Integer, MutableLong>(), (map, value) -> {
-                                MutableLong newValue = map.get(value) ;
-                                if (newValue == null) {
-                                    newValue = new MutableLong();
-                                    map.put(value, newValue);
-                                }
-                                newValue.incAndSet();
-                                return map;
-                            })
-                        .takeRight(1);
-                 ;
-
-        // number of blanks for a given letter
-        Function<Entry<Integer, MutableLong>, Long> blank =
-                entry ->
-                        Long.max(
-                            0L,
-                            entry.getValue().get() -
-                            scrabbleAvailableLetters[entry.getKey() - 'a']
-                        )
-                    ;
-
+                                               
+         // Histogram of the letters in a given word
+        Function<String, ReactiveSeq<Map.Entry<Integer, Long>>> histoOfLetters =
```

Comment:
Here is a list how you could change the original setup of operators:

- `ReactiveSeq<Integer> chars(String word)`: you can have a dedicated source that generates the characters
- `histoOfLetters` `scanLeft().takeRight()`: this can be a one step `collect()` operator
- `nBlanks` `scanLeft().takeRight()`: this can be a dedicated sum of longs operator
- `score2` `scanLeft().takeRight()`: this can be a dedicated sum of longs operator
- `toBeMaxed` `appendStream()`: can be changed to some `concat(s1, s2)` if that's faster for you
- `score3` `scanLeft().takeRight()`: this can be a dedicated sum of longs operator
- `buildHistoOnScore` `scanLeft().takeRight()`: this can be a one step `collect()` operator

> The benchmark you have published shows a 5X (or more) performance penalty for cyclops-react versus JDK 8 Streams, I have demonstrated clearly at this point that no such performance penalty exists.

What you demonstrated is that by not using ReactiveSeq, everything is faster. That means that since ReactiveSeq is a wrapper around Stream, you have overhead somewhere you don't see. With the same pattern, RxJava 2 gets really close to Java Stream because instead of complaining to the original author of the benchmarks, I took the time to optimize RxJava.

> I'll make the changes you have requested when we get equivalent changes to the Streams benchmark - otherwise I am accepting and endorsing a benchmark that is heavily biased in favour of Streams and against cyclops-react.

I think I already formulated my opinion about Cyclops-React. I you don't want to be mentioned in future benchmarks, I will remove this benchmark and not mention Cyclops-React unless I asked directly about it.